collie/metrics.py

```diff
@@ -8,33 +8,37 @@
 from torchmetrics import Metric
 from torchmetrics.functional import auroc
 from tqdm.auto import tqdm
+from typing_extensions import Literal
 
 import collie
 from collie.interactions import ExplicitInteractions, Interactions, InteractionsDataLoader
 from collie.model import BasePipeline
 
 
-def _get_user_item_pairs(user_ids: Union[np.array, torch.tensor],
-                         n_items: int,
+def _get_user_item_pairs(model: BasePipeline,
+                         user_or_item_ids: Union[np.array, torch.tensor],
+                         n_items_or_users: int,
                          device: Union[str, torch.device]) -> Tuple[torch.tensor, torch.tensor]:
     """
-    Create tensors pairing each input user ID with each item ID.
+    Create tensors pairing each input user ID with each item ID or vice versa.
 
     Parameters
     ----------
-    user_ids: np.array or torch.tensor, 1-d
-        Iterable[int] of users to score
-    n_items: int
-        Number of items in the training data
+    model: collie.model.BasePipeline
+        Model that can take a (user_id, item_id) pair as input and return a recommendation score
+    user_or_item_ids: np.array or torch.tensor, 1-d
```

Comment:
This is a time when I wish we had a Nicole around. For now, I think this works, but before we think about merging into the main Collie codebase, I think we should try to establish a better naming scheme that makes the code more readable while still understanding that things can refer to either users or items. I hate that we have to put `user_or_item` everywhere now, but I have no better ideas.

Reply:
Per offline discussion, `rows` and `cols` FTW!